packages/fluentui/react-component-event-listener/src/EventListenerWithComputedTarget.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+
+import useEventListener from './useEventListener';
+import { EventListenerOptions, EventTypes, Target } from './types';
+
+type EventListenerWithComputedTargetProps<T extends EventTypes> = Omit<EventListenerOptions<T>, 'target' | 'targetRef'> & {
+  resolveTarget: () => Target;
+};
+
+function EventListenerWithComputedTarget<T extends EventTypes>(props: EventListenerWithComputedTargetProps<T>) {
+  const { resolveTarget, ...rest } = props;
+  const targetRef = React.useRef<Target | null>(null);
+
+  React.useEffect(() => {
+    targetRef.current = resolveTarget();
```

Comment:
just to clarify my own understanding, in this effect hook, the return value of resolveTarget passed in from props is set to targetRef.current? if so, what is the [resolveTarget] doing below?

Reply:
We will use `resolveTarget()` as a delayed executor.

```tsx
  React.useEffect(() => {
    targetRef.current = resolveTarget();
  }, [resolveTarget]);
```

You're right, this `useEffect` will call `resolveTarget` and assign it to `targetRef`.

```tsx
useEventListener({ ...rest, targetRef });
```

`useEventListener` will use `targetRef`.

### How it's different?

`targetRef` will be resolved inside `EventListenerWithComputedTarget` instead of `List` and guarantees the proper priority of `ref` resolutions.